test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -577,6 +581,79 @@ def __set_test_end_data(self, test, err=None):
         test.duration = test.end_time - test.start_time
         test.captured_logs = self.result.handler.get_logs()
 
+    def __get_param_dictionary(self):
+        """
+        Method to convert data in param into dictionary of counts
+        """
+        paramkeys = ['server', 'servers', 'mom', 'moms', 'comms', 'client']
+        tparam_dic = {}
+        tparam_contents = {}
+        for key in paramkeys:
+            tparam_contents[key] = []
+        tparam_dic.update(default_requirements)
+        if self.param is None:
+            return tparam_dic
+        for h in self.param.split(','):
+            if '=' in h:
+                k, v = h.split('=')
+                if k in paramkeys:
+                    tparam_contents[k] = [v.replace(':', ',')]
+                    if (k == 'server' or k == 'servers'):
+                        tparam_dic['num_servers'] = len(v.split(':'))
+                    if (k == 'mom' or k == 'moms'):
+                        tparam_dic['num_moms'] = len(v.split(':'))
+                    if k == 'comms':
+                        tparam_dic['num_comms'] = len(v.split(':'))
+                    if k == 'clients':
+                        tparam_dic['num_clients'] = len(v.split(':'))
+        for pkey in paramkeys:
+            if pkey in ['server', 'mom', 'comms', 'client']:
+                if not tparam_contents[pkey]:
```

Comment:
since you are calling multiple time `set` on `tparam_contents` below, I will suggest to put else part here and modify `tparam_contents` value by calling `set` so that you don't have to call `set` below